### PR TITLE
Fix: Items

### PR DESCRIPTION
**Pull request by vorus**
```GetItemInfo``` on an item not in one's inventory returns nil.

Bug exists due to ```NeP.Core:GetItemID``` returning the argument (string) on failure to lookup item id.

Solution: Use the given item id if one is provided.  Attempt to get the item id otherwise.

### CODE REVIEW

File: system/compiler.lua [hunk 1]
```diff
@@ -91,11 +91,19 @@ function NeP.Compiler.Spell(eval, name)
...
 			end
 			if not ref.spell then return end
-			local itemName, itemLink, _,_,_,_,_,_,_, texture = GetItemInfo(ref.spell)
+			local itemID = tonumber(ref.spell)
+			if not itemID then
+				itemID = NeP.Core:GetItemID(itemName)
+			end
+			local itemName, itemLink, _,_,_,_,_,_,_, texture = GetItemInfo(itemID)
+			if not itemName or not tonumber(itemID) then
+				NeP.Core:Print('Failed to request item id for "' .. ref.spell .. '". Skipping.')
```
Comment (vorus) [inline system/compiler.lua]: Not sure if I should be printing errors in a combat rotation.

**Comment by vorus** [comment]
Should close #159 and #144.

File: system/compiler.lua (hunk 1 above)
Comment (modhunter) [inline system/compiler.lua]: I don't think you should either!

Comment (vorus) [inline system/compiler.lua]: I removed it.  Eventually there should be a log of all combat routine errors that are found.

**Pull request merged**